components/scream/src/mct_coupling/scream_cxx_f90_interface.cpp

```diff
@@ -80,6 +81,9 @@ void scream_create_atm_instance (const MPI_Fint& f_comm,
     // First of all, initialize the scream session
     scream::initialize_scream_session();
 
+    // Initialize yakl
+    if(!yakl::isInitialized()) { yakl::init(); }
```

Comment:
This might be personal, but I don't see why the cxx-f90 interface should do this explicitly. Imho, it should be done in one of the following places:

- inside `initialize_scream_session`: this is where also kokkos gets (indirectly) inited.
- inside rrtmgp interface (or any place using yakl): since rrtmgp is probably the only yakl customer (at the very least, they are not many), and since yakl offers the `isInitialized()` guard, maybe it should be up to rrtmgp (and possibly other yakl customers) to check/init yakl.

I guess the latter is more of a totalitarian kokkos regime approach, while the former is more of a peaceful coexistence and equal treatment approach. So I'm fine with the former, I guess.

Reply:
Question: does yakl need to be finalized as well?

Reply:
I agree it should be moved, and I think I like the Kooks-totalitarian approach of putting it into the interface since it is the only place that uses yakl..